corehq/apps/locations/forms.py

```diff
@@ -651,8 +653,11 @@ def __init__(self, domain, location, *args, **kwargs):
         self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
 
     def save(self):
-        selected_location_ids = set(self.cleaned_data['related_locations'].split(','))
-        previous_locations = self.location.related_location_ids
+        selected_location_ids = set()
+        if self.cleaned_data['related_locations']:
+            # ''.split(',') == ['']
+            selected_location_ids = set(self.cleaned_data['related_locations'].split(','))
```

Comment:
If the issue is that `self.cleaned_data['related_locations'] == ','`, then this wouldn't work, would it?  (`bool(',')` is `True`)

What about throwing a `filter(None, ...)` in there?

Reply:
the issue was when the return value was an empty string `''`. ',' shouldn't be returned from select2 widget, but fixed it here: https://github.com/dimagi/commcare-hq/pull/21431/commits/2820c93ca52252dd4bd2476c9ba859db15e56a8d